src/main/java/edu/hm/hafner/analysis/parser/QtTranslationSaxParser.java

```diff
@@ -0,0 +1,211 @@
+package edu.hm.hafner.analysis.parser;
+
+import java.util.ArrayDeque;
+import java.util.Deque;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.commons.lang3.StringUtils;
+import org.xml.sax.Attributes;
+import org.xml.sax.Locator;
+import org.xml.sax.helpers.DefaultHandler;
+
+import edu.hm.hafner.analysis.IssueBuilder;
+import edu.hm.hafner.analysis.ParsingException;
+import edu.hm.hafner.analysis.Report;
+import edu.hm.hafner.analysis.Severity;
+
+/**
+ * Handles the parsing of a translation file from Qt.
+ */
+public class QtTranslationSaxParser extends DefaultHandler {
+    private static final String CONTEXT = "context";
+    private static final String CONTEXT_NAME = "name";
+    private static final String MESSAGE = "message";
+    private static final String ROOT = "TS";
+    private static final String SOURCE = "source";
+    private static final String TRANSLATION = "translation";
+    private static final String TRANSLATION_ATTR_TYPE = "type";
+
+    static final String TRANSLATION_TYPE_OBSOLETE = "obsolete";
+    static final String TRANSLATION_TYPE_UNFINISHED = "unfinished";
+    static final String TRANSLATION_TYPE_VANISHED = "vanished";
+
+    static final String TRANSLATION_TYPE_OBSOLETE_MESSAGE =
+            "This translation can be removed as the source doesn't exists anymore.";
+    static final String TRANSLATION_TYPE_UNFINISHED_MESSAGE =
+            "This source string misses a translation.";
+    static final String TRANSLATION_TYPE_VANISHED_MESSAGE =
+            "The source string can not be found within the sources. "
+            + "Remove this translation if it is not used anymore or improve your call to \"tr()\", "
+            + "so \"lupdate\" can find it.";
+
+    private final Report report;
+    private final IssueBuilder builder = new IssueBuilder();
+    private Locator documentLocator;
+    private final Deque<String> elementTypeStack = new ArrayDeque<>();
+    private final Map<String, String> expectedElementTypeParents = new HashMap<>();
+    private String contextName = "";
+    private String sourceValue = "";
+    private boolean translationTagFound = false;
+    private boolean emitIssue = false;
+    private int lastColumnNumber;
+
+    /**
+     * Creates a new instance of {@link QtTranslationSaxParser}.
+     *
+     * @param report
+     *         the issues
+     * @param fileName
+     *         path to the translation file
+     */
+    public QtTranslationSaxParser(final Report report, final String fileName) {
```

Comment:
Please add a `@ SuppressFBWarnings ` annotation for `"NP_NONNULL_FIELD_NOT_INITIALIZED_IN_CONSTRUCTOR"` with the `justification` that it will be filled by SaxParser